File: go/texttest_fixture.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func main() {
	fmt.Println("OMGHAI!")

	var items = []*Item{
		{"+5 Dexterity Vest", 10, 20},
		{"Aged Brie", 2, 0},
		{"Elixir of the Mongoose", 5, 7},
		{"Sulfuras, Hand of Ragnaros", 0, 80},
		{"Sulfuras, Hand of Ragnaros", -1, 80},
		{"Backstage passes to a TAFKAL80ETC concert", 15, 20},
		{"Backstage passes to a TAFKAL80ETC concert", 10, 49},
		{"Backstage passes to a TAFKAL80ETC concert", 5, 49},
		// This Conjured item does not work properly yet
		{"Conjured Mana Cake", 3, 6}, // <-- :O
	}

	days := 2
	var err error
	if len(os.Args) > 1 {
		days, err = strconv.Atoi(os.Args[1])
		if err != nil {
			fmt.Println(err.Error())
			os.Exit(1)
		}
		days++
	}

	for day := 0; day < days; day++ {
		fmt.Printf("-------- day %d --------\n", day)
		fmt.Println("name, sellIn, quality")
		for i := 0; i < len(items); i++ {
			fmt.Printf("%s, %d, %d\n", items[i].name, items[i].sellIn, items[i].quality)
		}
		fmt.Println("")
		UpdateQuality(items)
	}
}

File: TypeScript/texttest-fixture.ts
import { Item, GildedRose } from './gilded-rose';

console.log('OMGHAI!');

const items = [
  new Item('+5 Dexterity Vest', 10, 20),
  new Item('Aged Brie', 2, 0),
  new Item('Elixir of the Mongoose', 5, 7),
  new Item('Sulfuras, Hand of Ragnaros', 0, 80),
  new Item('Sulfuras, Hand of Ragnaros', -1, 80),
  new Item('Backstage passes to a TAFKAL80ETC concert', 15, 20),
  new Item('Backstage passes to a TAFKAL80ETC concert', 10, 49),
  new Item('Backstage passes to a TAFKAL80ETC concert', 5, 49),
  // This Conjured item does not work properly yet
  new Item('Conjured Mana Cake', 3, 6), // <-- :O
];

let days = 2;
if (process.argv[2]) {
  days = parseFloat(process.argv[2]) + 1;
}

const gildedRose = new GildedRose(items);
for (let day = 0; day < days; day++) {
  console.log('-------- day %d --------', day);
  console.log('name, sellIn, quality');
  items.forEach(item => {
    console.log("%s, %d, %d", item.name, item.sellIn, item.quality);
  });
  console.log();
  gildedRose.updateQuality();
}
